### PR TITLE
Fix/uppsf 2749 migrate to cmneo4j

### PR DESCRIPTION
**Pull request by mchompalova**
# Description

## What

We migrated content-rw-neo4j to the new cm-neo4j-driver.

## Why

https://financialtimes.atlassian.net/browse/UPPSF-2749

## Scope and particulars of this PR (Please tick all that apply)

- [x] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: content/content_service.go [hunk 1]
```diff
@@ -160,60 +156,56 @@ func (cd service) Write(thing interface{}, transId string) error {
...
-	return err
+	return nil
 }
 
-func addStoryPackageRelationQuery(articleUUID, packageUUID string) *neoism.CypherQuery {
-	statement := `MERGE(sp:Thing{uuid:{packageUuid}})
+func addStoryPackageRelationQuery(articleUUID, packageUUID string) *cmneo4j.Query {
+	query := `MERGE(sp:Thing{uuid:{packageUuid}})
 			MERGE(c:Thing{uuid:{contentUuid}})
 			MERGE(c)<-[rel:IS_CURATED_FOR]-(sp)`
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service.go:182]: Fixed.

File: content/content_service.go [hunk 2]
```diff
@@ -160,60 +156,56 @@ func (cd service) Write(thing interface{}, transId string) error {
...
 	}
-	return query
 }
 
-func addContentPackageRelationQuery(articleUUID, packageUUID string) *neoism.CypherQuery {
-	statement := `MERGE(cp:Thing{uuid:{packageUuid}})
+func addContentPackageRelationQuery(articleUUID, packageUUID string) *cmneo4j.Query {
+	query := `MERGE(cp:Thing{uuid:{packageUuid}})
 			MERGE(c:Thing{uuid:{contentUuid}})
 			MERGE(c)-[rel:CONTAINS]->(cp)`
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service.go:196]: Fixed.

File: content/content_service_test.go [hunk 3]
```diff
@@ -554,98 +575,98 @@ func cleanDB(db neoutils.CypherRunner, t *testing.T, assert *assert.Assertions)
...
-func writeRelationship(db neoutils.NeoConnection, contentID string, conceptID string, t *testing.T, assert *assert.Assertions) {
-	var annotateQuery string
-	var qs []*neoism.CypherQuery
-
-	annotateQuery = `
+func writeRelationship(d *cmneo4j.Driver, contentID string, conceptID string, assert *assert.Assertions) {
+	annotateQuery := `
 		MERGE (content:Thing{uuid:{contentId}})
 		MERGE (concept:Thing) ON CREATE SET concept.uuid = {conceptId}
 		MERGE (content)-[pred:SOME_PPREDICATE]->(concept)
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service_test.go:632]: Fixed.

File: content/content_service_test.go [hunk 4]
```diff
@@ -554,98 +575,98 @@ func cleanDB(db neoutils.CypherRunner, t *testing.T, assert *assert.Assertions)
...
+		return false, nil
+	}
 	return len(result) > 0, err
 }
 
-func checkIsCuratedForRelationship(db neoutils.NeoConnection, spID string, assert *assert.Assertions) int {
+func checkIsCuratedForRelationship(d *cmneo4j.Driver, spID string, assert *assert.Assertions) int {
 	countQuery := `
 		MATCH (t:Thing{uuid:{storyPackageId}})-[r:IS_CURATED_FOR]->(x)
 		RETURN count(r) as c`
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service_test.go:672]: Fixed.

File: content/content_service_test.go [hunk 5]
```diff
@@ -654,19 +675,21 @@ func checkIsCuratedForRelationship(db neoutils.NeoConnection, spID string, asser
...
 	assert.NoError(err)
 
 	return results[0].Count
 }
 
-func checkContainsRelationship(db neoutils.NeoConnection, cpID string, assert *assert.Assertions) int {
+func checkContainsRelationship(d *cmneo4j.Driver, cpID string, assert *assert.Assertions) int {
 	countQuery := `
 		MATCH (t:Thing{uuid:{contentPackageId}})<-[r:CONTAINS]-(x)
 		RETURN count(r) as c`
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service_test.go:695]: Fixed.

File: content/content_service_test.go [hunk 6]
```diff
@@ -675,42 +698,45 @@ func checkContainsRelationship(db neoutils.NeoConnection, cpID string, assert *a
...
 	var result []struct {
 		UUID string `json:"t.uuid"`
 	}
 
-	checkGraph := neoism.CypherQuery{
-		Statement: `
+	checkGraph := &cmneo4j.Query{
+		Cypher: `
 			MATCH (t:Thing) WHERE t.uuid in {uuids}
 			RETURN t.uuid
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service_test.go:723]: Fixed.

File: content/content_service.go [hunk 7]
```diff
@@ -68,20 +70,20 @@ func (cd service) Read(uuid string, transId string) (interface{}, bool, error) {
...
 		Result: &results,
 	}
 
-	err := cd.conn.CypherBatch([]*neoism.CypherQuery{query})
+	err := cd.driver.Read(query)
 
-	if err != nil {
+	if errors.Is(err, cmneo4j.ErrNoResultsFound) {
 		return content{}, false, err
 	}
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service.go]: Fixed.

File: content/content_service_test.go [hunk 8]
```diff
@@ -423,27 +439,27 @@ func TestWriteNodeLabelsAreWrittenForContentPackage(t *testing.T) {
...
 			"uuid": genericContentPackage.UUID,
 		},
 		Result: &result,
 	}
 
-	err = contentDriver.conn.CypherBatch([]*neoism.CypherQuery{getNodeLabelsQuery})
+	err = d.Write(getNodeLabelsQuery)
 	assert.NoError(err)
 	assert.Len(result[0].NodeLabels, 3, "There should be 3 node labels: Thing, Content, ContentPackage")
 	assert.Equal("Thing", result[0].NodeLabels[0], "Thing should be the grandparent label")
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service_test.go]: Fixed.

File: content/content_service_test.go [hunk 9]
```diff
@@ -293,95 +307,97 @@ func TestUpdateWillRemovePropertiesNoLongerPresent(t *testing.T) {
...
+			Result: &result,
 		},
-		Result: &result,
 	}
 
-	err = contentDriver.conn.CypherBatch([]*neoism.CypherQuery{getNodeLabelsQuery})
+	err = d.Write(getNodeLabelsQuery...)
 	assert.NoError(err)
 	assert.Len(result[0].NodeLabels, 2, "There should be 2 node labels: Thing, Content")
 	assert.Equal("Thing", result[0].NodeLabels[0], "Thing should be the parent label")
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply content/content_service_test.go]: Fixed.

File: content/content_service_test.go [hunk 10]
```diff
@@ -554,163 +572,168 @@ func cleanDB(db neoutils.CypherRunner, t *testing.T, assert *assert.Assertions)
...
-func writeRelationship(db neoutils.NeoConnection, contentID string, conceptID string, t *testing.T, assert *assert.Assertions) {
-	var annotateQuery string
-	var qs []*neoism.CypherQuery
-
-	annotateQuery = `
-		MERGE (content:Thing{uuid:{contentId}})
+func writeRelationship(d *cmneo4j.Driver, contentID string, conceptID string, assert *assert.Assertions) {
+	annotateQuery := `
+		MERGE (content:Thing{uuid:$contentId})
 		MERGE (concept:Thing) ON CREATE SET concept.uuid = {conceptId}
```
Comment (ivanruski) [inline content/content_service_test.go:628]: This one should be updated as well.

> ```suggestion

		MERGE (concept:Thing) ON CREATE SET concept.uuid = $conceptId
```

**Review by ivanruski — CHANGES_REQUESTED** [review]
Just one more thing needs to be changed.

**Review by ivanruski — DISMISSED** [review]
LGTM

**Review by ManoelMilchev — APPROVED** [review]

**Review by ManoelMilchev — DISMISSED** [review]
LGTM

**Review by ManoelMilchev — APPROVED** [review]

**Review by ivanruski — APPROVED** [review]

**Pull request merged**